dags/utils/callbacks.py

```diff
@@ -0,0 +1,26 @@
+from typing import Optional
+
+from airflow.models.dagrun import DagRun
+from airflow.models.taskinstance import clear_task_instances
+from airflow.utils.context import Context
+from airflow.utils.db import provide_session
+from sqlalchemy.orm.session import Session
+
+
+@provide_session
+def retry_tasks_callback(context: Context, session: Optional[Session] = None) -> None:
+    """Callback to clear tasks specified by the `retry_tasks` task param.
+    
+    Intended to be used to as an `on_retry_callback` to also retry other tasks when a task fails.
+    """
+    retry_task_ids: list[str] = context['params'].get('retry_tasks', [])
```

Comment:
I also considered having the task IDs passed in as a function argument, but that would require slightly more complex usage with something like [`functools.partial()`](https://docs.python.org/3.8/library/functools.html?highlight=partial#functools.partial) (and having to import that to use it), and a side benefit of doing it with task params is those will get captured as task instance metadata so we can see what the `retry_tasks` param values were after the fact.

Reply:
> ... a side benefit of doing it with task params is those will get captured as task instance metadata so we can see what the `retry_tasks` param values were after the fact.

This turned out to be untrue, as Airflow unhelpfully displays the `params` attribute value as a Python object reference (e.g. `<airflow.models.param.ParamsDict object at 0x7f15cfd69cd0>`).